runtime/commonMain/src/kotlinx/serialization/json/internal/StreamingJsonInput.kt

```diff
@@ -108,6 +109,19 @@ internal class StreamingJsonInput internal constructor(
         }
     }
 
+    private fun treatAsMissing(descriptor: SerialDescriptor, index: Int): Boolean {
+        if (!configuration.treatNullAsMissing) return false
+        val elementDescriptor = descriptor.getElementDescriptor(index)
+        if (reader.tokenClass == TC_NULL && !elementDescriptor.isNullable) return true // null for non-nullable
+        if (elementDescriptor.kind == UnionKind.ENUM_KIND) {
+            val enumValue = reader.peekString(configuration.isLenient)
+                    ?: return false // if value is not a string, decodeEnum() will throw correct exception
+            val enumIndex = elementDescriptor.getElementIndex(enumValue)
+            if (enumIndex == UNKNOWN_NAME) return true
```

Comment:
`return enumIndex == UNKNOWN_NAME` is a bit more straightforward

Reply:
I'd prefer to stick with the current version because it's easier to spot places where the function returns true, and the only place where it returns false is at the end. But I can change this if you insist.